src/main/java/com/sparta/spartaSimulator/model/TraineeCentre.java

```diff
@@ -46,8 +66,20 @@ public void checkCentreStatus() {
         }
     }
 
+    //Changes to this method to check specialism against centres
     public void addTrainee(Trainee trainee){
-        allTrainees.add(trainee);
+        String specialism = "";
+        try
+        {
+            specialism = getCentreSpecialism().toString();
+        } catch (NullPointerException ex)
+        {
+            allTrainees.add(trainee);
+        }
+        if(trainee.getTraineeCourse().toString().equals(specialism))
+        {
+            allTrainees.add(trainee);
+        }
```

Comment:
Does not look for other random trainees with the specified course. If the techCentre wants 13 trainees and there are 13 trainees in the waiting list with the required course, no guarantee they will be sent to centre, as there is no condition to pick another if they have the wrong course.